### PR TITLE
Mosip-25203 develop

### PR DESCRIPTION
**Pull request by AkshayChouguleac1**

### CODE REVIEW

**Comment by AkshayChouguleac1** [comment]
Resolved sonar cloud bugs mentioned at 
https://sonarcloud.io/project/issues?resolved=false&types=BUG&id=mosip_registration

for 
https://sonarcloud.io/project/issues?resolved=false&types=BUG&id=mosip_registration&open=AYUbC2XoUnZyzNW2Nf01
used try with resources to close the stream

for
https://sonarcloud.io/project/issues?resolved=false&types=BUG&id=mosip_registration&open=AX95VOR5DrMh0EA-JQMd
there was a comment given already
 // Sonar detecting default value and estimating condition as always false but in our case default value can be also null
this is invalid bug

File: registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java [hunk 1]
```diff
@@ -634,14 +634,15 @@ private boolean uinUpdate(String regId, String process, String uin, MessageDTO o
 		} else {
 			String statusComment = result != null && result.getErrors() != null ? result.getErrors().get(0).getMessage()
 					: UINConstants.NULL_IDREPO_RESPONSE;
+			String message = result != null && result.getErrors() != null
```
Comment (MonobikashDas) [inline registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java:637]: use !CollectionUtils.isEmpty()
It will do both null and empty check.

File: registration-processor/core-processor/registration-processor-demo-dedupe-stage/src/main/java/io/mosip/registration/processor/stages/demodedupe/DemodedupeProcessor.java [hunk 2]
```diff
@@ -358,7 +358,7 @@ private DemoDedupeStatusDTO insertDemodedupDetailsAndPerformDedup(IndividualDemo
 			}
 			else {
 				String  demo=env.getProperty(DEMODEDUPEENABLE);
-				if (StringUtils.isNotEmpty(demo) && demo.trim().equalsIgnoreCase(TRUE)) {
```
Comment (MonobikashDas) [inline registration-processor/core-processor/registration-processor-demo-dedupe-stage/src/main/java/io/mosip/registration/processor/stages/demodedupe/DemodedupeProcessor.java:361]: are we using apache - commons lang StringUtils.isNotEmpty(demo)?. It does both null and empty check. Why this change is required?

File: registration-processor/registration-processor-rest-client/src/main/java/io/mosip/registration/processor/rest/client/utils/RestApiClient.java [hunk 3]
```diff
@@ -227,8 +228,11 @@ private HttpEntity<Object> setRequestHeader(Object requestType, MediaType mediaT
 				Iterator<String> iterator = httpHeader.keySet().iterator();
 				while (iterator.hasNext()) {
 					String key = iterator.next();
-					if (!(headers.containsKey("Content-Type") && key.equalsIgnoreCase("Content-Type")) && httpHeader.get(key).size()>0)
-						headers.add(key,httpHeader.get(key).get(0));
+					List<String> collection = httpHeader.get(key);
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-rest-client/src/main/java/io/mosip/registration/processor/rest/client/utils/RestApiClient.java:231]: why have you changed this logic?

**Review by MonobikashDas — CHANGES_REQUESTED** [review]

File: registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java (hunk 1 above)
Comment (AkshayChouguleac1) [reply registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java:637]: Logically StringUtils.isEmpty() checks for null also and previous development was correct but sonar shows it as a bug . after doing null check != null , It is resolved that's why did like this.

Comment (AkshayChouguleac1) [reply registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java:637]: Logically StringUtils.isNotEmpty() checks for null also and previous development was correct but sonar shows it as a bug . after doing null check != null , It is resolved that's why did like this.

File: registration-processor/core-processor/registration-processor-demo-dedupe-stage/src/main/java/io/mosip/registration/processor/stages/demodedupe/DemodedupeProcessor.java (hunk 2 above)
Comment (AkshayChouguleac1) [reply registration-processor/core-processor/registration-processor-demo-dedupe-stage/src/main/java/io/mosip/registration/processor/stages/demodedupe/DemodedupeProcessor.java:361]: Logically StringUtils.isNotEmpty() checks for null also and previous development was correct but sonar shows it as a bug . after doing null check != null , It is resolved that's why did like this.

File: registration-processor/registration-processor-rest-client/src/main/java/io/mosip/registration/processor/rest/client/utils/RestApiClient.java (hunk 3 above)
Comment (AkshayChouguleac1) [reply registration-processor/registration-processor-rest-client/src/main/java/io/mosip/registration/processor/rest/client/utils/RestApiClient.java:231]: Sonar was showing bug for previos development (httpHeader.get(key) can return null ) , that's why modifiend in such a way that it will be resolved and logic will not change ..

Comment (AkshayChouguleac1) [reply registration-processor/registration-processor-rest-client/src/main/java/io/mosip/registration/processor/rest/client/utils/RestApiClient.java:231]: Many of the changes have already been merged in previous pr, only changes in LandingZoneUtility.java and StageClassesUtil.java are yet to be merged

File: registration-processor/core-processor/registration-processor-demo-dedupe-stage/src/main/java/io/mosip/registration/processor/stages/demodedupe/DemodedupeProcessor.java (hunk 2 above)
Comment (AkshayChouguleac1) [reply registration-processor/core-processor/registration-processor-demo-dedupe-stage/src/main/java/io/mosip/registration/processor/stages/demodedupe/DemodedupeProcessor.java:361]: Many of the changes have already been merged in previous pr, only changes in LandingZoneUtility.java and StageClassesUtil.java are yet to be merged

File: registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java (hunk 1 above)
Comment (AkshayChouguleac1) [reply registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java:637]: Many of the changes have already been merged in previous pr, only changes in LandingZoneUtility.java and StageClassesUtil.java are yet to be merged

**Comment by MonobikashDas** [comment]
@AkshayChouguleac1 please resolve merge conflicts.
